SwiftUIDZ/SwiftUIDZ/CarsView.swift

```diff
@@ -0,0 +1,235 @@
+    @State private var segmentIndexRentalPeriod = 0
+    @State private var toggleIsOn = false
+    @State private var progress: Double = 0
+    @State private var isAlertShow = false
+    @State private var isShareShow = false
+    @State private var offset = 0
+    
+    // MARK: - Private properties
+    
+    private var cars = Car.getCars()
+    private var rents = Rent.getRens()
+
+    // MARK: - Body
+    
+    var body: some View {
+        NavigationView {
+            VStack {
+                ZStack(alignment: .bottom) {
+                    ZStack(alignment: .top) {
+                        Circle()
+                            .fill(
+                                LinearGradient(
+                                    gradient: Gradient(colors: [Color(UIColor.systemGray5), .black]),
+                                    startPoint: .top, endPoint: .bottom
+                                )
+                            )
+                            .frame(width: Constants.bigImageSize, height: Constants.bigImageSize)
+                        Text(cars[segmentIndex].name)
+                            .fontWeight(.bold)
+                            .font(Font.custom(Constants.fontName, size: Constants.fontSizeBig).weight(.bold))
+                            .foregroundColor(.white)
+                            .padding()
+                            .offset(x: CGFloat(offset))
+                    }
+                    Image(cars[segmentIndex].name)
+                        .resizable()
+                        .scaledToFit()
+                        .offset(x: CGFloat(offset))
+                }
+                .animation(.spring(response: 0.3, dampingFraction: 0.7, blendDuration: 1))
+                HStack(spacing: Constants.smallSpacing) {
+                    createCarParameter(
+                        imageName: Constants.meterImageName,
+                        parameterText: cars[segmentIndex].speed
+                    )
+                    createCarParameter(
+                        imageName: Constants.seatImageName,
+                        parameterText: cars[segmentIndex].seats
+                    )
+                    createCarParameter(
+                        imageName: Constants.engineImageName,
+                        parameterText: cars[segmentIndex].power
+                    )
+                }
+                .padding()
+                .frame(height: Constants.commonSize)
+                .foregroundColor(Color(UIColor.systemGray6))
+                createCarNameSegment()
+                createForm()
+                Text(Constants.distanceText)
+                    .font(Font.custom(Constants.fontName, size: Constants.fontSizeRegular).weight(.bold))
+                    .foregroundColor(.white)
+                createDistanceSlider()
+                HStack(spacing: Constants.spacing) {
+                    Text(toggleIsOn ? "\(Constants.totalText) \(rents[segmentIndexRentalPeriod].price + 100) $" :
```

Comment:
```suggestion
                    Text(toggleIsOn ? "\(Constants.totalText) \(rents[segmentIndexRentalPeriod].price + 100) $" :
```

100 - магическое число